### PR TITLE
Pprevops 813 implement iframe tracking to templates

### PR DESCRIPTION
**Pull request by kbigos**
I've added Iframe communication to most of the templates. Changed the structure of sending values outside so that it would be the same for all the templates which have the Iframe tracking option so that only one handler will be needed. The handler was also added. We send the following in the message data - {type: "PiwikPRO", payload: eventData}  We later use the type to recognise messages from our templates and convert them into events with our handler. 

It can be tested here, not all templates but a few (I tested all of them) - https://multiwp.demopiwikpro.com/kbigos/form-test-iframe/

Dead clicks - click 2 times on something 
Rage clicks - click 3 times on something 
Copied text - copy something 
Form Tracking - click on the search input and click on something else

### CODE REVIEW

**Review by rezik007 — COMMENTED** [review]
Proposed solution implies that:
1. IframeTracking logic is duplicated in multiple templates
2. This logic must be applied to all new templates
3. This logic affects core logic of each template

I would highly recommend to provide solution which at least doesn't affect templates codes and their interfaces so it could look more or less like this (dead click, templates.js file):
```
${fs.readFileSync(path.join(__dirname, 'build/withPostMessage.js'), { encoding: 'utf-8' })}
${fs.readFileSync(path.join(__dirname, 'build/detectDeadClicks.js'), { encoding: 'utf-8' })}

var unsubscribe = detectDeadClicks(function (eventData) {
var trackFromIframe = {{iframeTracking}};
if (!trackFromIframe) {
  window._paq.push(eventData);
} else {
  withPostMessage(eventData);
}

// unsubscribe();
}, {
interval: {{interval}},
limit: {{limit}},
});
```

So to do that we would need that function exposed (withPostMessage.js or something):
```
export default (payload) => {
  window.parent.postMessage({ type: 'PiwikPRO', payload }, '*');
};
```

But even little better solution would be to unify callback functions whereever we can so every code could look like:
```
${fs.readFileSync(path.join(__dirname, 'build/pushToAnalytics.js'), { encoding: 'utf-8' })}
${fs.readFileSync(path.join(__dirname, 'build/someFunction.js'), { encoding: 'utf-8' })}

var unsubscribe = someFunction(function (eventData) {
pushToAnalytics(eventData);

// unsubscribe();
}, {
prop: {{propValue}}
});
```

However that can be out of scope of this task. I only think how to inform users that whenever their choose "from iframe tracking" option they also have to register that new template with listener to get things work.... 🤔

File: templates.js [hunk 1]
```diff
@@ -21,18 +21,25 @@ module.exports = {
...
+iframeTracking: {{iframeTracking}},
 });
       `,
       arguments: [
         { id: 'interval', type: 'number', displayName: 'Time interval', description: 'Number of milliseconds to reset counter', default: 1000 },
         { id: 'limit', type: 'number', displayName: 'Number of clicks needed to trigger subscription', default: 2 },
+        { id: 'iframeTracking',
+        type: 'boolean',
+        displayName: 'Send messages from iframes',
+        description: 'If checked, you won’t send the _paq.push but instead you will send a message',
```
Comment (RomanSek) [inline templates.js:40]:
```suggestion
        description: 'If checked, you won’t send the _paq.push but instead you will send a message to the parent window',
```

**Review by RomanSek — APPROVED** [review]

**Comment by kbigos** [comment]
@rezik007  @RomanSek  I've included your proposed changes in the latest commits. Please have a look whenever you can :)

File: src/detectQuickBacks.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default (subscribe, { threshold = 12000 }) => {
 
   document.addEventListener('click', listener);
 
-  return () => {
+ /* return () => {
```
Comment (RomanSek) [inline src/detectQuickBacks.js:27]: Should it be commented out?

Comment (kbigos) [reply src/detectQuickBacks.js]: Must have left it after testing, removed the comments in the latest commit. Thanks for spotting that :)

**Review by rezik007 — APPROVED** [review]
Code looking fine, but I didn't tested iframe setup in practice (I assume you did it).

**Pull request merged**